core/domain/recommendations_services.py

```diff
@@ -340,3 +352,36 @@ def get_exploration_recommendations(exp_id):
         return []
     else:
         return recommendations_model.recommended_exploration_ids
+
+
+def delete_explorations_from_recommendations(exp_ids):
+    """Deletes explorations from recommendations.
+
+    This deletes both the recommendations for the given explorations, as well as
+    the given explorations from other explorations' recommendations.
+
+    Args:
+        exp_ids: list(str). List of exploration IDs for which to delete
+            the recommendations.
+    """
+    recs_model_class = (
+        recommendations_models.ExplorationRecommendationsModel)
+    recommendation_models = recs_model_class.get_multi(exp_ids)
+    existing_recommendation_models = [
+        model for model in recommendation_models if model is not None]
+    recs_model_class.delete_multi(existing_recommendation_models)
+
+    all_recommending_models = {}
+    for exp_id in exp_ids:
+        recommending_models = recs_model_class.query(
+            recs_model_class.recommended_exploration_ids == exp_id
+        ).fetch()
+        for recommending_model in recommending_models:
+            all_recommending_models[recommending_model.id] = (
```

Comment:
You never use the ID, AFAICT. So ... why not have all_recommending_models just be a list of model instances, instead?

Reply:
Added an explanatory comment above.